public/components/FeatureAnywhereContextMenu/CreateAnomalyDetector/AddAnomalyDetector.tsx

```diff
@@ -0,0 +1,823 @@
+import { DataFilterList } from '../../../../public/pages/DefineDetector/components/DataFilterList/DataFilterList';
+import {
+  convertTimestampToNumber,
+  getError,
+  isInvalid,
+  validateDetectorName,
+  validateFeatureName,
+  validatePositiveInteger,
+} from '../../../../public/utils/utils';
+import { CUSTOM_AD_RESULT_INDEX_PREFIX } from '../../../../server/utils/constants';
+import {
+  focusOnFirstWrongFeature,
+  initialFeatureValue,
+} from '../../../../public/pages/ConfigureModel/utils/helpers';
+import { FeaturesFormikValues } from '../../../../public/pages/ConfigureModel/models/interfaces';
+import {
+  getIndices,
+  getMappings,
+} from '../../../../public/redux/reducers/opensearch';
+import {
+  featuresToUIMetadata,
+  formikToFeatureAttributes,
+  formikToFilterQuery,
+} from '../../../../public/pages/ReviewAndCreate/utils/helpers';
+import { FormattedFormRow } from '../../../../public/components/FormattedFormRow/FormattedFormRow';
+import { FeatureAccordion } from '../../../../public/pages/ConfigureModel/components/FeatureAccordion';
+import {
+  AD_DOCS_LINK,
+  MAX_FEATURE_NUM,
+} from '../../../../public/utils/constants';
+
+function AddAnomalyDetector({
+  embeddable,
+  closeFlyout,
+  notification,
+  services,
+  mode,
+  setMode,
+  index,
+  selectedDetectorId,
+  setSelectedDetectorId,
+}) {
+  const dispatch = useDispatch();
+  const [queryText, setQueryText] = useState('');
+  useEffect(() => {
+    const getInitialIndices = async () => {
+      await dispatch(getIndices(queryText));
+    };
+    getInitialIndices();
+    dispatch(getMappings(embeddable.vis.data.aggs.indexPattern.title));
+  }, []);
+
+  const [isShowVis, setIsShowVis] = useState(false);
+  const [accordionsOpen, setAccordionsOpen] = useState({ modelFeatures: true });
+  const [detectorNameFromVis, setDetectorNameFromVis] = useState(
+    formikToDetectorName(embeddable.vis.title)
+  );
+  const [intervalValue, setIntervalalue] = useState(10);
+  const [delayValue, setDelayValue] = useState(1);
+  const [enabled, setEnabled] = useState<boolean>(false);
+  const [shingleSizeValue, setShingleSizeValue] = useState(8);
+
+  const title = embeddable.getTitle();
+  const onAccordionToggle = (key) => {
+    const newAccordionsOpen = { ...accordionsOpen };
+    newAccordionsOpen[key] = !accordionsOpen[key];
+    setAccordionsOpen(newAccordionsOpen);
+  };
+  const onDetectorNameChange = (e, field) => {
+    field.onChange(e);
+    setDetectorNameFromVis(e.target.value);
+  };
+  const onIntervalChange = (e, field) => {
+    field.onChange(e);
+    setIntervalalue(e.target.value);
+  };
+  const onDelayChange = (e, field) => {
+    field.onChange(e);
+    setDelayValue(e.target.value);
+  };
+  const aggList = embeddable.vis.data.aggs.aggs.filter(
+    (feature) => feature.schema == 'metric'
+  );
+  const featureList = aggList.filter(
+    (feature, index) => index < (aggList.length < 5 ? aggList.length : 5)
+  );
```

Comment:
Was there any UX decision on displaying any messaging if metrics limit > feature limit? 

Also, we should actually be pulling the feature limit from cluster settings instead of hardcoded to 5, but that will be an overall improvement since we have it hardcoded elsewhere too so no blocker on that part.

Reply:
Actually, I see you have `MAX_FEATURE_NUM` constant at least, should use that

Reply:
There wasn't any UX decision on displaying any messaging, I will double check Xenia on this.